contrib/drivers/ceph/replication.go

```diff
@@ -0,0 +1,407 @@
+
+	volumename := opensdsPrefix + opt.PrimaryVolumeId
+
+	cephconfig := &ssh.ClientConfig{
+		User: r.conf.Username,
+		Auth: []ssh.AuthMethod{
+			ssh.Password(r.conf.Password),
+		},
+		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
+	}
+	cephclient, err := ssh.Dial("tcp", r.conf.HostdailIP, cephconfig)
+	if err != nil {
+		log.Error("Failed to dial: " + err.Error())
+
+	}
+
+	backupconfig := &ssh.ClientConfig{
+		User: r.conf.Username,
+		Auth: []ssh.AuthMethod{
+			ssh.Password(r.conf.Password),
+		},
+		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
+	}
+	backupclient, error := ssh.Dial("tcp", r.conf.PeerdailIP, backupconfig)
+
+	if error != nil {
+		log.Error("Failed to dial: " + error.Error())
+
+	}
+
+	cephenablesession, err := cephclient.NewSession()
+	if err != nil {
+	}
+
+	cephenablesession.Run("rbd  feature enable rbd/" + volumename + " exclusive-lock,journaling")
+	cephenablesession.Close()
+
+	cephrbdinstall, err := cephclient.NewSession()
+	if err := cephrbdinstall.Run("apt install -y rbd-mirror"); err != nil {
+		log.Error("Failed to run: " + err.Error())
+
+	}
+	cephrbdinstall.Close()
+
+	backupsession6, err := backupclient.NewSession()
+
+	if err := backupsession6.Run("apt install -y rbd-mirror"); err != nil {
+		log.Error("Failed to run: " + err.Error())
+
+	}
+	backupsession6.Close()
+
+	cephauthsession, err := cephclient.NewSession()
+	if err != nil {
+		log.Error("Failed to create session: " + err.Error())
+
+	}
+
+	if err := cephauthsession.Run("ceph auth get-or-create client.ceph mon 'allow r' osd 'allow class-read object_prefix rbd_children, allow rwx pool=rbd' -o /etc/ceph/ceph.client.ceph.keyring"); err != nil {
+		log.Error("Failed to run: " + err.Error())
+	}
+	cephauthsession.Close()
+
+	backupauthsession, error := backupclient.NewSession()
+	if error != nil {
+		log.Error("Failed to create session: " + err.Error())
+
+	}
+
+	if err := backupauthsession.Run("ceph --cluster remote auth get-or-create client.remote mon 'allow r' osd 'allow class-read object_prefix rbd_children, allow rwx pool=rbd' -o /etc/ceph/remote.client.remote.keyring"); err != nil {
+		log.Error("Failed to run: " + err.Error())
+	}
+	backupauthsession.Close()
+
+	cephenablesession, err = cephclient.NewSession()
+	if err != nil {
+
+	}
+
+	if err := cephenablesession.Run("rbd mirror pool enable rbd image"); err != nil {
+		log.Error("Failed to run: " + err.Error())
+	}
+	cephenablesession.Close()
+
+	backupenablesession, error := backupclient.NewSession()
+	if error != nil {
+	}
+
+	if err := backupenablesession.Run("rbd --cluster remote mirror pool enable rbd image"); err != nil {
+		log.Error("Failed to run: " + err.Error())
+	}
+	backupenablesession.Close()
+
+	cephscpsession, err := cephclient.NewSession()
+	if err != nil {
+		log.Error("Failed to create session: " + err.Error())
+
+	}
+	cmd := "scp /etc/ceph/ceph.client.ceph.keyring /etc/ceph/ceph.conf root@" + r.conf.Ipaddresspeer + ":/etc/ceph/"
```

Comment:
why are all CEPH related paths hardcoded ? please remove. we have 3 options here 
1. read CEPH env. variables
2. or read CEPH conf. files for details
3. or define our own conf files